style: docs/ polish (#5234)
prefix-dev/pixi: docs/python/pytorch.md

```

### Mixing MacOS and CUDA with `pypi-dependencies`

When using pypi-dependencies, Pixi creates a “solve” environment to resolve the PyPI dependencies.
This process involves installing the Conda dependencies first and then resolving the PyPI packages within that environment.

This can become problematic if you’re on a macOS machine and trying to resolve the CUDA version of PyTorch for Linux or Windows.

> Since macOS doesn’t support these operating systems, the Conda dependencies for CUDA will fail to install, preventing proper resolution.

Reviewer: ```suggestion
Since macOS doesn’t support the Conda dependencies for CUDA, it can't install the solve environment, preventing proper resolution.
```

This is something highly specific the the pypi solving. And it's not "always" true but I really don't know how to explain that in a few lines anymore.
